### PR TITLE
OBSDOCS-460: Clean up collector docs

### PR DESCRIPTION
**Pull request by abrennan89**
Version(s):
4.11+

Issue:
https://issues.redhat.com/browse/OBSDOCS-460

Link to docs preview:
- https://66628--docspreview.netlify.app/openshift-enterprise/latest/logging/cluster-logging (moved content out, clean up)
- https://66628--docspreview.netlify.app/openshift-enterprise/latest/logging/log_collection_forwarding/log-forwarding

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: logging/log_collection_forwarding/log-forwarding.adoc [hunk 1]
```diff
@@ -7,6 +7,71 @@ include::_attributes/attributes-openshift-dedicated.adoc[]
...
+
+* `kafka`. A Kafka broker. The `kafka` output can use an unsecured or TLS connection.
+
+* `loki`. Loki, a horizontally scalable, highly available, multitenant log aggregation system.
+
+.Outputs
+[options="header"]
+|==========================================================
+| Feature              | Fluentd  | Vector
+| Elasticsearch v5-v7  | &#10003; | &#10003;
```
Comment (anpingli) [inline logging/log_collection_forwarding/log-forwarding.adoc:62]: This should be Elasticsearch v6-v8

File: logging/log_collection_forwarding/log-forwarding.adoc [hunk 2]
```diff
@@ -7,6 +7,71 @@ include::_attributes/attributes-openshift-dedicated.adoc[]
...
+| Fluent forward       | &#10003; |
+| Syslog RFC3164       | &#10003; | &#10003; (Logging 5.7+)
+| Syslog RFC5424       | &#10003; | &#10003; (Logging 5.7+)
+| Kafka                | &#10003; | &#10003;
+| Cloudwatch           | &#10003; | &#10003;
+| Loki                 | &#10003; | &#10003;
+| HTTP                 | &#10003; | &#10003; (Logging 5.7+)
```
Comment (anpingli) [inline logging/log_collection_forwarding/log-forwarding.adoc:69]: Splunk is supported from 5.6 https://issues.redhat.com/browse/LOG-2913

File: logging/log_collection_forwarding/log-forwarding.adoc (hunk 1 above)
Comment (anpingli) [reply logging/log_collection_forwarding/log-forwarding.adoc:62]: Google Cloud Logging is supported from 5.5.1 https://issues.redhat.com/browse/LOG-2574

File: logging/log_collection_forwarding/log-forwarding.adoc [hunk 3]
```diff
@@ -7,6 +7,71 @@ include::_attributes/attributes-openshift-dedicated.adoc[]
...
+.Outputs
+[options="header"]
+|==========================================================
+| Feature              | Fluentd  | Vector
+| Elasticsearch v5-v7  | &#10003; | &#10003;
+| Fluent forward       | &#10003; |
+| Syslog RFC3164       | &#10003; | &#10003; (Logging 5.7+)
+| Syslog RFC5424       | &#10003; | &#10003; (Logging 5.7+)
+| Kafka                | &#10003; | &#10003;
+| Cloudwatch           | &#10003; | &#10003;
```
Comment (anpingli) [inline logging/log_collection_forwarding/log-forwarding.adoc:67]: Cloudwatch sts is supported from  5.5. https://issues.redhat.com/browse/LOG-1976

File: logging/log_collection_forwarding/log-forwarding.adoc [hunk 4]
```diff
@@ -7,6 +7,71 @@ include::_attributes/attributes-openshift-dedicated.adoc[]
...
+[id="log-forwarding-collection"]
+== Log collection
+
+link:https://vector.dev/docs/about/what-is-vector/[Vector] is a log collector offered as an alternative to Fluentd for the {logging}.
+
+You can configure which logging collector type your cluster uses by modifying the `ClusterLogging` custom resource (CR) `collection` spec.
+
+[IMPORTANT]
+====
+Vector does not support FIPS Enabled Clusters.
```
Comment (anpingli) [inline logging/log_collection_forwarding/log-forwarding.adoc:23]: Vector support FIPS now  https://issues.redhat.com/browse/LOG-2460

File: logging/log_collection_forwarding/log-forwarding.adoc [hunk 5]
```diff
@@ -7,6 +7,71 @@ include::_attributes/attributes-openshift-dedicated.adoc[]
...
+
+.Example ClusterLogging CR that configures Vector as the collector
+[source,yaml]
+----
+  apiVersion: "logging.openshift.io/v1"
+  kind: ClusterLogging
+  metadata:
+    name: instance
+    namespace: openshift-logging
+    annotations:
```
Comment (anpingli) [inline logging/log_collection_forwarding/log-forwarding.adoc:34]: It is not preview-vector-collector  from 5.5. We can remove this annotations

File: modules/logging-vector-fluentd-feature-comparison.adoc [hunk 6]
```diff
@@ -95,8 +95,3 @@ include::snippets/logging-outputs-5.6-snip.adoc[]
 | Log event buffering   | &#10003; |
```
Comment (anpingli) [inline modules/logging-vector-fluentd-feature-comparison.adoc:95]: Lokistack on IPv6  are fixed recently https://issues.redhat.com/browse/LOG-4570. Let me confirm if that had been backport to 5.5.

Comment (abrennan89) [reply modules/logging-vector-fluentd-feature-comparison.adoc:95]: I'm not sure this is really within the scope of the PR?

**Review by jcantrill — COMMENTED** [review]
/lgtm

File: modules/about-log-collection.adoc [hunk 7]
```diff
@@ -0,0 +1,53 @@
+// Module included in the following assemblies:
+//
+// * logging/log_collection_forwarding/log-forwarding.adoc
+
+:_content-type: CONCEPT
```
Comment (michaelryanpeter) [inline modules/about-log-collection.adoc:5]:
```suggestion
:_mod-docs-content-type: CONCEPT
```

File: modules/log-forwarding-collector-outputs.adoc [hunk 8]
```diff
@@ -0,0 +1,26 @@
+// Module included in the following assemblies:
+//
+// * logging/log_collection_forwarding/log-forwarding.adoc
+
+:_content-type: REFERENCE
```
Comment (michaelryanpeter) [inline modules/log-forwarding-collector-outputs.adoc:5]:
```suggestion
:_mod-docs-content-type: REFERENCE
```

File: modules/logging-vector-fluentd-feature-comparison.adoc [hunk 9]
```diff
@@ -1,11 +1,10 @@
...
+// Module included in the following assemblies:
 //
-:_mod-docs-content-type: REFERENCE
-[id="logging-5-6-collector-ref_{context}"]
+// * logging/log_collection_forwarding/log-forwarding.adoc
 
-= Collector features
-
-include::snippets/logging-outputs-5.6-snip.adoc[]
+:_content-type: REFERENCE
```
Comment (michaelryanpeter) [inline modules/logging-vector-fluentd-feature-comparison.adoc:5]:
```suggestion
:_mod-docs-content-type: REFERENCE
```

File: modules/about-log-collection.adoc [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    logs:
+      type: vector
+      vector: {}
+# ...
+----
+
+[id="about-log-collectors-limitations_{context}"]
+== Log collection limitations
+
+The container runtimes provide minimal information to identify the source of log messages: project, pod name, and container ID. This information is not sufficient to uniquely identify the source of the logs. If a pod with a given name and project is deleted before the log collector begins processing its logs, information from the API server, such as labels and annotations, might not be available. There might not be a way to distinguish the log messages from a similarly named pod and project or trace the logs to their source. This limitation means that log collection and normalization are considered *best effort*.
```
Comment (michaelryanpeter) [inline modules/about-log-collection.adoc:46]: ISG specifies using italics for emphasized words: https://www.ibm.com/docs/en/ibm-style?topic=format-highlighting#highlighting-by-element

File: modules/about-log-collection.adoc [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+[IMPORTANT]
+====
+The available container runtimes provide minimal information to identify the
+source of log messages and do not guarantee unique individual log
+messages or that these messages can be traced to their source.
```
Comment (michaelryanpeter) [inline modules/about-log-collection.adoc:52]: Avoid hard wrapping lines.

**Review by michaelryanpeter — COMMENTED** [review]
A few nits. Otherwise, LGTM. Great work.

/label peer-review-done
/remove-label peer-review-in-progress
/remove-label peer-review-needed

**Comment by abrennan89** [comment]
@michaelryanpeter thank you so much! ❤️

**Pull request merged**